tools/unit_test_runner.sh

```diff
@@ -36,6 +36,20 @@ if [[ -z $VT_GO_PARALLEL && -n $VT_GO_PARALLEL_VALUE ]]; then
   VT_GO_PARALLEL="-p $VT_GO_PARALLEL_VALUE"
 fi
 
+# Mac makes long temp directories for os.TempDir(). MySQL can't connect to
+# sockets in those directories. Tell Golang to use /tmp/vttest_XXXXXX instead.
+kernel="$(uname -s)"
+case "$kernel" in
+  darwin|Darwin)
+    TMPDIR=${TMPDIR:-}
+    if [ -z "$TMPDIR" ]; then
+      TMPDIR="$(mktemp -d /tmp/vttest_XXXXXX)"
```

Comment:
I think we should still use a pseudo unique dir, no? We could instead use:
```
TMPDIR="$(mktemp -d /tmp/vttest_$RANDOM)"
```

Or am I missing something?

Reply:
@mattlord `TMPDIR="$(mktemp -d /tmp/vttest_XXXXXX)"` produces unique dirs. `mktemp` replaces the `X`s with random chars. On my machine:

```
~ mktemp -d /tmp/vttest_XXXXXX
/tmp/vttest_EKyPRE
```

Let me know if you think that's that's good-enough-uniqueness.

FYI I had no idea about `$RANDOM` -- that's cool!

Reply:
Ah, nice! TiL. I do vaguely recall seeing that before now that you mention it... That's good enough then (it's even more pseudo random/unique)!